Assets/Plugins/iOS/Shopify/MessageDispatcher.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  Dispatcher.swift
+//  UnityBuySDK
+//
+//  Created by Shopify.
+//  Copyright © 2017 Shopify Inc. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+protocol MessageDispatcher {
+    typealias DispatchCompletion<T: Deserializable> = (_ response: T?) -> Void
+    var receiver: String { get }
```

Comment:
Instead of a 1-1 per class, this abstraction is more like 1-1 per `GameObject`. It makes it safer because it removes the possibility of making a mistake where you mismatch the `GameObject` name and the methods that they handle. In Unity a `GameObject` holds multiple `components`, which are classes, at any moment these components can be added or removed. 

In my thought process, I thought that it would be more familiar to the Unity developer if each `component` was represented by `Dispatcher`